### ISSUE TITLE
The Direct module exposes way too much

### ISSUE THREAD

**Issue by treeowl** [simonmar/monad-par#62]
I think it's pretty weird that `Direct.Par` derives a `MonadReader Sched` instance. It's also pretty weird that `Direct` exposes the `Sched` type despite not offering any documentation of it or any functions for working with it. I think it probably makes sense to remove that instance and require users to import `DirectInternal` to get the `Sched` type.

**Comment by simonmar** [comment]
Agreed, and I definitely support efforts to clean up and document the API here.
